commit 6fc8dbfb8c38ef314139fa126d8c902c982e90f9
Author: Shanker Donthineni <shankerd@codeaurora.org>
Date:   Tue Mar 7 08:20:38 2017 -0600

    irqchip/gicv3-its: Add workaround for QDF2400 ITS erratum 0065
    
    commit 90922a2d03d84de36bf8a9979d62580102f31a92 upstream.
    
    On Qualcomm Datacenter Technologies QDF2400 SoCs, the ITS hardware
    implementation uses 16Bytes for Interrupt Translation Entry (ITE),
    but reports an incorrect value of 8Bytes in GITS_TYPER.ITTE_size.
    
    It might cause kernel memory corruption depending on the number
    of MSI(x) that are configured and the amount of memory that has
    been allocated for ITEs in its_create_device().
    
    This patch fixes the potential memory corruption by setting the
    correct ITE size to 16Bytes.
    
    Cc: stable@vger.kernel.org
    Signed-off-by: Shanker Donthineni <shankerd@codeaurora.org>
    Signed-off-by: Marc Zyngier <marc.zyngier@arm.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/arch/arm64/Kconfig
===================================================================
--- linuxtest.orig/arch/arm64/Kconfig
+++ linuxtest/arch/arm64/Kconfig
@@ -484,6 +484,16 @@ config QCOM_QDF2400_ERRATUM_0065
 
 	  If unsure, say Y.
 
+config QCOM_QDF2400_ERRATUM_0065
+	bool "QDF2400 E0065: Incorrect GITS_TYPER.ITT_Entry_size"
+	default y
+	help
+	  On Qualcomm Datacenter Technologies QDF2400 SoC, ITS hardware reports
+	  ITE size incorrectly. The GITS_TYPER.ITT_Entry_size field should have
+	  been indicated as 16Bytes (0xf), not 8Bytes (0x7).
+
+	  If unsure, say Y.
+
 endmenu
 
 
Index: linuxtest/dummy/rpi_1453_6fc8dbfb8c38ef314139fa126d8c902c982e90f9.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1453_6fc8dbfb8c38ef314139fa126d8c902c982e90f9.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
